### ISSUE TITLE
HtmlUnit -> Error when loading JavascriptConfiguration.xml

### ISSUE THREAD

**Issue by ghost** [adamdoupe/enemy-of-the-state#2]
Hi Adam,

I am sadly unable to run the enemy of the state :-[ I am using Jython and I did checkout the proper branch.

This is the error I get :

```
com.gargoylesoftware.htmlunit.javascript.configuration.JavaScriptConfiguration loadConfiguration
SEVERE: Error when loading JavascriptConfiguration.xml
java.lang.NullPointerException 
at com.gargoylesoftware.htmlunit.javascript.configuration.JavaScriptConfiguration.getConfigurationFileAsReader(JavaScriptConfiguration.java:257)
......
```

Thanks !

**Comment by adamdoupe** [comment]
Humm, I haven't seen that. I was able to run it on my machine.

Here's my jython version numer: Jython 2.5.3

And my java version:
java version "1.7.0_45"
OpenJDK Runtime Environment (IcedTea 2.4.3) (ArchLinux build 7.u45_2.4.3-1-x86_64)
OpenJDK 64-Bit Server VM (build 24.45-b08, mixed mode)

I ran this:
export JYTHONPATH=./lib/apache-mime4j-0.6.jar:./lib/commons-codec-1.4.jar:./lib/commons-collections-3.2.1.jar:./lib/commons-io-1.4.jar:./lib/commons-lang-2.4.jar:./lib/commons-logging-1.1.1.jar:./lib/cssparser-0.9.5.jar:./lib/htmlunit-2.8.jar:./lib/htmlunit-core-js-2.8.jar:./lib/httpclient-4.0.1.jar:./lib/httpcore-4.0.1.jar:./lib/httpmime-4.0.1.jar:./lib/nekohtml-1.9.14.jar:./lib/sac-1.3.jar:./lib/serializer-2.7.1.jar:./lib/xalan-2.7.1.jar:./lib/xercesImpl-2.9.1.jar:./lib/xml-apis-1.3.04.jar

Then this:
jython crawler2.py "http://google.com"                                            

and it seemed to work.

What OS are you using?

**Comment by ghost** [comment]
I am using Ubuntu 12.04 i686.
my java version :
java version "1.7.0_45"
Java(TM) SE Runtime Environment (build 1.7.0_45-b18)
Java HotSpot(TM) Server VM (build 24.45-b08, mixed mode)

JYTHONPATH is properly set.

I did a quick search and what always come up is that you might have forgotten to include /src/main/resources in the classpath when you compiled HtmlUnit. But then again, I don't know why you can make it work and I don't...

**Comment by adamdoupe** [comment]
If you unzip lib/htmlunit-2.8.jar there's a file:
com/gargoylesoftware/htmlunit/javascript/configuration/JavaScriptConfiguration.xml

Can you verify if you're seeing the same thing or not?

**Comment by ghost** [comment]
Yep I can see the file :/

**Comment by adamdoupe** [comment]
I still don't know why you're seeing that, but maybe you can try this:

unzip every file in lib/*.jar to the root repo directory

So you should have these directories under enemy-of-the-state:
audit/  com/  java_cup/  javax/  jcc/  lib/  license/  META-INF/  net/  org/  test/  trax/

Java may be able to pick up the files from there.

If that doesn't work, can you post the debug.log file from the following strace:
strace -f -o debug.log jython crawler2.py "http://google.com"

**Comment by ghost** [comment]
Well, unzipping all JARs did not do the trick.

Since the debug.log file weights more than 23 MB, I've sent you an email with a download link.

Thanks again.

**Comment by adamdoupe** [comment]
Very strange, I still don't know what's happening there.

The only difference right now is that I'm using OpenJDK.

Can you use that instead of Oracle Java? That way we can eliminate one of the variables.

**Comment by ghost** [comment]
Erf, I am still have the same issue.

java version "1.7.0_25"
OpenJDK Runtime Environment (IcedTea 2.3.10) (7u25-2.3.10-1ubuntu0.12.04.2)
OpenJDK Server VM (build 23.7-b01, mixed mode)

FYI, right before getting the NullPointerException, I get this : 

```
INFO:Engine:starting with URL 1/1 http://google.com
janv. 15, 2014 7:48:06 PM com.gargoylesoftware.htmlunit.WebClient printContentIfNecessary
INFO: statusCode=[301] contentType=[text/html]
janv. 15, 2014 7:48:06 PM com.gargoylesoftware.htmlunit.WebClient printContentIfNecessary
INFO: <HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>301 Moved</TITLE></HEAD><BODY>
<H1>301 Moved</H1>
The document has moved
<A HREF="http://www.google.com/">here</A>.
</BODY></HTML>
```

I always get a 301, regardless of the URL I provided.

**Comment by adamdoupe** [comment]
OK Alex, I may have figured out a workaround.

I was able to reproduce your problem on a fresh install of Ubuntu 12.04.

However, Ubuntu 13.10 seems to work.

My current theory is that there's something that was fixed in either the Jython or Java to fix this problem (which is not in my patch, but in the HtmlUnit code).

The exception is a null pointer exception on this line:
http://sourceforge.net/p/htmlunit/code/HEAD/tree/tags/HtmlUnit-2.8/src/main/java/com/gargoylesoftware/htmlunit/javascript/configuration/JavaScriptConfiguration.java#l257

I have no idea how to fix it, but if you manage to figure it out I will gladly accept a pull request.

At this point, all I'm going to do is update the README so that people will know what works.
